client/app/components/DynamicComponent.jsx

```diff
@@ -21,6 +21,19 @@ export function unregisterComponent(name) {
   registerComponent(name, null);
 }
 
+export function isComponentRegistered(name) {
+  return isFunction(componentsRegistry.get(name));
+}
+
+export function isComponentAvailable(name) {
+  const component = componentsRegistry.get(name);
+  if (isFunction(component)) {
+    const result = invoke(component, "isComponentAvailable");
+    return isUndefined(result) ? true : !!result;
+  }
+  return false;
+}
+
```

Comment:
@gabrieldutra This one shouldn't be ported - I'm gonna remove this in internal fork as well